lib/csetgrp.gi

```diff
@@ -573,21 +573,20 @@ end);
 InstallMethod(RightCoset,"use subgroup size",IsCollsElms,
   [IsGroup and HasSize,IsObject],0,
 function(U,g)
-local d,fam;
+local d,fam,typ;
   # noch tests...
 
   fam:=FamilyObj(U);
-  if not IsBound(fam!.rightCosetsDefaultSizeType) then
-    fam!.rightCosetsDefaultSizeType:=NewType(fam,IsRightCosetDefaultRep and
-          HasActingDomain and HasFunctionAction and HasRepresentative and
-	  HasSize and HasCanonicalRepresentativeDeterminatorOfExternalSet);
-  fi;
+  typ:=NewType(fam,IsRightCosetDefaultRep and
+          HasActingDomain and HasFunctionAction and HasRepresentative
+	  and HasCanonicalRepresentativeDeterminatorOfExternalSet);
 
   d:=rec();
-  ObjectifyWithAttributes(d,fam!.rightCosetsDefaultSizeType,
+  ObjectifyWithAttributes(d,typ,
     ActingDomain,U,FunctionAction,OnLeftInverse,Representative,g,
-    Size,Size(U),CanonicalRepresentativeDeterminatorOfExternalSet,
+    CanonicalRepresentativeDeterminatorOfExternalSet,
     RightCosetCanonicalRepresentativeDeterminator);
+  SetSize(d,Size(U)); # as own setter
```

Comment:
What does "as own setter" mean?

Also, doesn't `ObjectifyWithAttributes` call custom setters anyway? It contains specific code to detect them. But perhaps you want to avoid that? In that case, I guess a comment explaining that (and why) would be indeed appropriate, but this comment isn't doing that.

Reply:
We cannot set the size in the previous `ObjectifyWithAttributes` as there is a custom setter method (the one just added here). In such a case `ObjectifyWith Attributes just does `Objectify` and calls all setters separately which is what we want to avoid here.